tensorflow_datasets/image/fruits360.py

```diff
@@ -0,0 +1,125 @@
+volume = {10},
+journal = {Acta Universitatis Sapientiae, Informatica},
+doi = {10.2478/ausi-2018-0002}
+}
+"""
+
+_COMMIT_SHA = 'b6960fc0287be349c222c3dbdd2624958c902431'
+_DOWNLOAD_URL = 'https://github.com/Horea94/Fruit-Images-Dataset/archive/{sha}.tar.gz'.format(sha=_COMMIT_SHA)
+
+_IMAGE_SIZE = 100
+_IMAGE_SHAPE = (_IMAGE_SIZE, _IMAGE_SIZE, 3)
+_CLASS_NAMES = ['Apple Braeburn', 'Apple Crimson Snow', 'Apple Golden 1', 'Apple Golden 2', 'Apple Golden 3',
+                'Apple Granny Smith', 'Apple Pink Lady', 'Apple Red 1', 'Apple Red 2', 'Apple Red 3',
+                'Apple Red Delicious', 'Apple Red Yellow 1', 'Apple Red Yellow 2', 'Apricot', 'Avocado', 'Avocado ripe',
+                'Banana', 'Banana Lady Finger', 'Banana Red', 'Cactus fruit', 'Cantaloupe 1', 'Cantaloupe 2',
+                'Carambula', 'Cherry 1', 'Cherry 2', 'Cherry Rainier', 'Cherry Wax Black', 'Cherry Wax Red',
+                'Cherry Wax Yellow', 'Chestnut', 'Clementine', 'Cocos', 'Dates', 'Granadilla', 'Grape Blue',
+                'Grape Pink', 'Grape White', 'Grape White 2', 'Grape White 3', 'Grape White 4', 'Grapefruit Pink',
+                'Grapefruit White', 'Guava', 'Hazelnut', 'Huckleberry', 'Kaki', 'Kiwi', 'Kohlrabi', 'Kumquats', 'Lemon',
+                'Lemon Meyer', 'Limes', 'Lychee', 'Mandarine', 'Mango', 'Mangostan', 'Maracuja', 'Melon Piel de Sapo',
+                'Mulberry', 'Nectarine', 'Orange', 'Papaya', 'Passion Fruit', 'Peach', 'Peach 2', 'Peach Flat', 'Pear',
+                'Pear Abate', 'Pear Kaiser', 'Pear Monster', 'Pear Red', 'Pear Williams', 'Pepino', 'Pepper Green',
+                'Pepper Red', 'Pepper Yellow', 'Physalis', 'Physalis with Husk', 'Pineapple', 'Pineapple Mini',
+                'Pitahaya Red', 'Plum', 'Plum 2', 'Plum 3', 'Pomegranate', 'Pomelo Sweetie', 'Quince', 'Rambutan',
+                'Raspberry', 'Redcurrant', 'Salak', 'Strawberry', 'Strawberry Wedge', 'Tamarillo', 'Tangelo',
+                'Tomato 1', 'Tomato 2', 'Tomato 3', 'Tomato 4', 'Tomato Cherry Red', 'Tomato Maroon', 'Tomato Yellow',
+                'Walnut']
+
+
+class Fruits360(tfds.core.GeneratorBasedBuilder):
+  """Fruits 360 dataset."""
+
+  VERSION = tfds.core.Version("1.0.0", experiments={tfds.core.Experiment.S3: True})
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description="A large set of fruits on a white background.",
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(shape=_IMAGE_SHAPE),
+            "image/filename": tfds.features.Text(),
+            "label": tfds.features.ClassLabel(names=_CLASS_NAMES)
+        }),
+        supervised_keys=("image", "label"),
+        urls=["https://www.kaggle.com/moltean/fruits"],
+        citation=_CITATION
+    )
+
+  def _split_generators(self, dl_manager):
+    resource = tfds.download.Resource(url=_DOWNLOAD_URL, extract_method=tfds.download.ExtractMethod.TAR_GZ)
```

Comment:
why did you specifically define this? `dl_manager` automatically finds the extract method, isn't it?

Reply:
Not in this case for some unknown reason.

Reply:
It may have to do with how GitHub's archive download endpoints work, I'm not really sure.

It doesn't hurt to explicitly set it.